api/envoy/api/v2/core/protocol.proto

```diff
@@ -27,6 +27,14 @@ message HttpProtocolOptions {
   // Note that request based timeouts mean that HTTP/2 PINGs will not keep the connection alive.
   google.protobuf.Duration idle_timeout = 1;
 
+  // The maximum lifetime of downstream connection. The lifetime timeout is defined
+  // as a period since downstream connection was established.
+  // If not set, there is no lifetime timeout. When lifetime_timeout is reached the connection will
+  // be closed. If the connection is an HTTP/2 connection a drain sequence will occur prior to
+  // closing the connection. See :ref:`drain_timeout
+  // <envoy_api_field_config.filter.network.http_connection_manager.v2.HttpConnectionManager.drain_timeout>`.
+  google.protobuf.Duration lifetime_timeout = 3;
```

Comment:
Nit: probably bike shedding, but I think I prefer `connection_maximum_duration` as more descriptive.

Reply:
renamed to `max_connection_duration`